src/app/model/Queen.ts

```diff
@@ -0,0 +1,50 @@
+import { PiecePositions } from './PiecesPositions';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { columns, Row, Side, Square, Vector } from './Types';
+import { PiecePossibleMoves } from './PiecePossibleMoves';
+
+export class Queen extends Piece implements PiecePossibleMoves {
+  constructor(side: Side) {
+    super(side);
+  }
+
+  possibleMoves(position: Square, board: PiecePositions): Square[] {
+    return [
+      ...this.lineMoves(board, position, { col: 0, row: -1 }),
+      ...this.lineMoves(board, position, { col: 0, row: 1 }),
+      ...this.lineMoves(board, position, { col: -1, row: 0 }),
+      ...this.lineMoves(board, position, { col: 1, row: 0 }),
+      ...this.lineMoves(board, position, { row: -1, col: -1 }),
+      ...this.lineMoves(board, position, { row: -1, col: 1 }),
+      ...this.lineMoves(board, position, { row: 1, col: 1 }),
+      ...this.lineMoves(board, position, { row: 1, col: -1 }),
+    ];
+  }
+
+  private lineMoves(board: PiecePositions, actualPosition: Square, vector: Vector): Square[] {
+    const nextSquare: Square = {
```

Comment:
To nie jest odniesienie to tego kodu, ale mi bardzo przeszkadza, ze Square nie jest klasa. Bo 100 razy powtarzamy kwesite typu: columns[columns.indexOf(actualPosition.column) + vector.col]
Moznaby zrobic metode w Square i zaimplementowac to raz :) 
Ale nie wiem czy to nie za duza zmiana. Ewentualnie jako dodatkowe zadanie to moze wejsc.

Np. wtedy bylaby metoda square.transform(vector) i milibysmy co chcemy :) To by moglo pilnowac tez kwestii czy nie wychodzimy poza board (nawet!)